lib/dialects/mssql/query-generator.js

```diff
@@ -322,7 +322,7 @@ const QueryGenerator = {
     return sql;
   },
 
-  upsertQuery(tableName, insertValues, updateValues, where, rawAttributes) {
+  upsertQuery(tableName, insertValues, updateValues, where, model) {
```

Comment:
Not strictly as it's undocumented. I know I've used `QueryGenerator.selectQuery()` sometimes to generate subqueries. But I guess that would be my own risk